### PR TITLE
[Perf] Retarget to .NET Core 2.0 and remove unnecessary assemblies

### PR DESCRIPTION
**Pull request by nguerrera**
This removes 2 MB of files that were being copied unnecessarily to the netcoreapp folder of the sdk. They were:

1. duplicates of shared framework assemblies (because dependencies pulled in 4.3.x/1.1 assemblies, but we targeted 1.0)

2. nuget assemblies that are loaded from CLI directory

3. core-setup dependencies that are loaded from CLI directory

I'm doing this as step 1 to crossgen'ing the netcoreapp variant of the SDK. I didn't want to have to crossgen and sign the extra unnecessary files, which would slow down the build and make the bloat even bigger.

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Extensions.Tasks/Microsoft.NET.Build.Extensions.Tasks.csproj [hunk 1]
```diff
@@ -30,9 +30,8 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Build.Framework" Version="$(MsBuildPackagesVersion)" ExcludeAssets="Runtime" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MsBuildPackagesVersion)" ExcludeAssets="Runtime" />
-    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" Condition=" '$(TargetFramework)' == 'netcoreapp1.0' " />
-    <PackageReference Include="System.Diagnostics.FileVersionInfo" Version="4.0.0" Condition=" '$(TargetFramework)' == 'netcoreapp1.0' " />
-    <PackageReference Include="XliffTasks" Version="$(XliffTasksVersion)" />
+    <PackageReference Include="XliffTasks" Version="$(XliffTasksVersion)" PrivateAssets="All" />
+    <PackageReference Include="NETStandard.Library.NETFramework" Version="$(NETStandardLibraryNETFrameworkVersion)" ExcludeAssets="All" NoWarn="NU1701" />
```
Comment (livarcocc) [inline src/Tasks/Microsoft.NET.Build.Extensions.Tasks/Microsoft.NET.Build.Extensions.Tasks.csproj]: Why is this added here?

**Review by livarcocc — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Extensions.Tasks/Microsoft.NET.Build.Extensions.Tasks.csproj (hunk 1 above)
Comment (nguerrera) [reply src/Tasks/Microsoft.NET.Build.Extensions.Tasks/Microsoft.NET.Build.Extensions.Tasks.csproj]: It's moved from the other project. This is the one that actually needs to restore it.

Comment (livarcocc) [reply src/Tasks/Microsoft.NET.Build.Extensions.Tasks/Microsoft.NET.Build.Extensions.Tasks.csproj]: I should have searched for it. I would have noticed the removal. I even looked for it and didn't see. Which is why I asked. I will go ahead and practice my reading.

Comment (nguerrera) [reply src/Tasks/Microsoft.NET.Build.Extensions.Tasks/Microsoft.NET.Build.Extensions.Tasks.csproj]: It's an unrelated change that should have been a separate commit. Shame on me for not following my own advice.
